general/polls/cog.py

```diff
@@ -53,137 +147,554 @@ async def send_poll(
+        embed_message = await channel.fetch_message(poll.message_id)
+        interaction_message = await channel.fetch_message(poll.interaction_message_id)
+    except NotFound:
+        poll.active = False
+        return
+
+    await interaction_message.delete()
+    embed = embed_message.embeds[0]
+    embed.set_footer(text=t.footer_closed)
+
+    await embed_message.edit(embed=embed)
+    await embed_message.unpin()
+
+    poll.active = False
+
+
+class MySelect(Select):
+    @db_wrapper
+    async def callback(self, interaction):
+        user = interaction.user
+        selected_options: list = self.values
+        message: Message = await interaction.channel.fetch_message(interaction.custom_id)
+        embed: Embed = message.embeds[0] if message.embeds else None
+        poll: Poll = await db.get(Poll, (Poll.options, Option.votes), message_id=message.id)
+        if not poll or not embed:
+            return
+
+        options: list[Option] = poll.options
+        new_options: list[Option] = [option for option in options if option.option in selected_options]
+        missing: list[Member] | None = None
+
+        opt: Option
+        for opt in poll.options:
+            for vote in opt.votes:
+                if vote.user_id == user.id:
+                    await vote.remove()
+                    opt.votes.remove(vote)
+
+        ev_pover = await PollsDefaultSettings.everyone_power.get()
+        if poll.fair:
+            user_weight: float = ev_pover
+        else:
+            highest_role = await RoleWeight.get_highest(user.roles) or 0
+            user_weight: float = ev_pover if highest_role < ev_pover else highest_role
+        for option in new_options:
+            option.votes.append(
+                await PollVote.create(option_id=option.id, user_id=user.id, poll_id=poll.id, vote_weight=user_weight)
+            )
+        if poll.poll_type == "team":
+            teamlers: set[Member] = await get_teamler(interaction.guild, ["team"])
+            if user not in teamlers:
+                await interaction.response.send_message(content=t.team_yn_poll_forbidden, ephemeral=True)
+                return
+
+            user_ids: set[int] = set()
+            for option in poll.options:
+                for vote in option.votes:
+                    user_ids.add(vote.user_id)
+
+            missing: list[Member] | None = [teamler for teamler in teamlers if teamler.id not in user_ids]
+            missing.sort(key=lambda m: str(m).lower())
+
+        embed = await edit_poll_embed(embed, poll, missing)
+        await message.edit(embed=embed)
 
 
 class PollsCog(Cog, name="Polls"):
-    CONTRIBUTORS = [Contributor.MaxiHuHe04, Contributor.Defelo, Contributor.TNT2k, Contributor.wolflu]
+    CONTRIBUTORS = [
+        Contributor.MaxiHuHe04,
+        Contributor.Defelo,
+        Contributor.TNT2k,
+        Contributor.wolflu,
+        Contributor.NekoFanatic,  # rewrote most of this code (Please blame @Defelo for the code)
```

Comment:
```suggestion
        Contributor.NekoFanatic,  
``` Should not be in productive code

Reply:
just change it.